tests/modules/images/webp-uploads/webp-uploads-test.php

```diff
@@ -385,4 +385,163 @@ public function it_should_remove_the_webp_version_of_the_image_if_the_image_is_f
+		yield 'An image with a .jpg extension' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg' );
+		yield 'An image with a .jpeg extension' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/car.jpeg' );
+	}
+
+	/**
+	 * Contain the full image size from the original mime
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_contain_the_full_image_size_from_the_original_mime() {
+		$attachment_id = $this->factory->attachment->create_upload_object(
+			TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg'
+		);
+
+		$tag = wp_get_attachment_image( $attachment_id, 'full', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+
+		$expected = array(
+			'ext'  => 'jpg',
+			'type' => 'image/jpeg',
+		);
+		$this->assertSame( $expected, wp_check_filetype( get_attached_file( $attachment_id ) ) );
+		$this->assertContains( wp_basename( get_attached_file( $attachment_id ) ), webp_uploads_update_image_references( $tag ) );
+	}
+
+	/**
+	 * Prevent replacing an image with no available sources
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_prevent_replacing_an_image_with_no_available_sources() {
+		add_filter( 'webp_uploads_supported_image_mime_transforms', '__return_empty_array' );
+
+		$attachment_id = $this->factory->attachment->create_upload_object( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/car.jpeg' );
+
+		$tag      = wp_get_attachment_image( $attachment_id, 'full', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+		foreach ( $metadata['sizes'] as $size => $properties ) {
+			$this->assertArrayNotHasKey( 'sources', $properties );
+		}
+
+		$this->assertSame( $tag, webp_uploads_update_image_references( $tag ) );
+	}
+
+	/**
+	 * Prevent update not supported images with no available sources
+	 *
+	 * @dataProvider data_provider_not_supported_webp_images
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_prevent_update_not_supported_images_with_no_available_sources( $image_path ) {
+		$attachment_id = $this->factory->attachment->create_upload_object( $image_path );
+
+		$this->assertIsNumeric( $attachment_id );
+		$tag      = wp_get_attachment_image( $attachment_id, 'full', false, array( 'class' => "wp-image-{$attachment_id}" ) );
+		$metadata = wp_get_attachment_metadata( $attachment_id );
+		foreach ( $metadata['sizes'] as $size => $properties ) {
+			$this->assertArrayNotHasKey( 'sources', $properties );
+		}
+
+		$this->assertSame( $tag, webp_uploads_update_image_references( $tag ) );
+	}
+
+	public function data_provider_not_supported_webp_images() {
+		yield 'PNG image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/dice.png' );
+		yield 'GIFT image' => array( TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/earth.gif' );
+	}
```

Comment:
I think this test is unnecessary. It tests exactly the same as the above `it_should_prevent_replacing_an_image_with_no_available_sources()`, which is that an image without `sources` is not being replaced. Whether there are no `sources` because of the filter returning an empty value or because the image is a PNG/GIF is irrelevant to `webp_uploads_update_image_references()` - if anything, that should be tested for `webp_uploads_generate_image_size()` or `webp_uploads_get_supported_image_mime_transforms()` which are actually responsible for that.

Reply:
Good point, the main reason for PNG and GIFt was to make sure those URL extensions were not affected by the regex inside of the function that does the replacement. Which is slightly different from `it_should_prevent_replacing_an_image_with_no_available_sources` which mimics an image that was uploaded before the plugin and module was activated.